### PR TITLE
Sphinx docs build

### PR DESCRIPTION
**Pull request by laserkelvin**
# ALCHEMI Toolkit Pull Request

## Description

Adds buildable sphinx docs

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Performance improvement
- [x] Documentation update
- [x] Refactoring (no functional changes)
- [ ] CI/CD or infrastructure change

## Changes Made

- Adds structure for dynamics, models, and data for users
- Adds guides on how users can build their own dynamics, wrap their own models, and use data structures
- Replaces `forces_are_conservative` in the model card with `forces_via_autograd` instead

## Testing

- [ ] Unit tests pass locally (`make pytest`)
- [ ] Linting passes (`make lint`)
- [ ] New tests added for new functionality meets coverage expectations?

## Checklist

- [ ] I have read and understand the [Contributing Guidelines](../CONTRIBUTING.md)
- [ ] I have updated the [CHANGELOG.md](../CHANGELOG.md)
- [ ] I have performed a self-review of my code
- [ ] I have added docstrings to new functions/classes
- [ ] I have updated the documentation (if applicable)

## Additional Notes

> [!TIP]
> This repository uses Greptile, an AI code review service, to help conduct
> pull request reviews. We encourage contributors to read and consider suggestions
> made by Greptile, but note that human maintainers will provide the necessary
> reviews for merging: Greptile's comments are **not** a qualitative judgement
> of your code, nor is it an indication that the PR will be accepted/rejected.
> We encourage the use of emoji reactions to Greptile comments, depending on
> their usefulness and accuracy.

### CODE REVIEW

File: docs/userguide/data.md [hunk 1]
```diff
@@ -98,6 +123,177 @@ list_of_data = batch.to_data_list()
...
+| ASE source | AtomicData field | Notes |
+|-------------------------------|----------------------|-----------------------------------------------|
+| `atoms.numbers` | `atomic_numbers` | |
+| `atoms.positions` | `positions` | |
+| `atoms.get_pbc()` | `pbc` | Reshaped to `(1, 3)` |
+| `atoms.get_cell()` | `cell` | Reshaped to `(1, 3, 3)` |
+| `atoms.info[energy_key]` | `energies` | Default key: `"energy"` |
+| `atoms.arrays[forces_key]` | `forces` | Default key: `"forces"` |
+| `atoms.info[stress_key]` | `stresses` | Voigt vector converted to `(3, 3)` matrix |
+| `atoms.info[virials_key]` | `virials` | Voigt vector converted to `(3, 3)` matrix |
```
Comment (dallasfoster) [inline docs/userguide/data.md]: Should be (1, 3, 3)?

File: docs/userguide/dynamics_simulations.md [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+        velocities = batch.velocities
+        forces = batch.forces
+        masses = batch.atomic_masses.unsqueeze(-1)
+
+        with torch.no_grad():
+            new_acc = forces / masses
+            if self._prev_accelerations is not None:
+                velocities.add_(0.5 * (self._prev_accelerations + new_acc) * self.dt)
+            else:
+                velocities.add_(new_acc * self.dt)
```
Comment (dallasfoster) [inline docs/userguide/dynamics_simulations.md:210]: Is this code block tested? I think we need to ensure that every code block in the documentation has been tested to some degree.
